Ignore node_modules when checking Python quality

andy-armstrong opened the pull request:
In preparation for the upcoming FedX work, this change updates PEP8 and PyLint to exclude anything beneath node_modules. I also added docs and .pycharm_helpers (for PyCharm users).

@nedbat @benpatterson Please review.

andy-armstrong commented on .pep8:
```diff
@@ -0,0 +1,4 @@
+[pep8]
```
Reviewers: I borrowed this from the ecommerce .pep8 file:

https://github.com/edx/ecommerce/blob/master/.pep8

benpatterson commented:
I think we should keep docs in. There's not a lot, but there is python in there.

andy-armstrong commented:
@benpatterson Thanks, that makes sense. I'll keep docs in.

andy-armstrong commented:
@benpatterson Does it make sense that the approach should be that we exclude directories containing code that we install, rather than code that lives in the repo. We exclude node_modules and .pycharm_helpers because those are directories where code is installed.

Anyway, I've removed docs and squashed the commits. Please review again.

andy-armstrong commented:
In fact, migrations is generated code, so that's another example that makes sense to exclude.

andy-armstrong commented:
I've added ``.git`` to the exclusion list too, since that was failing on Jenkins (although not locally for me for some reason).

benpatterson commented:
@andy-armstrong Yes I agree with the approach. 

Changes :+1: from me

andy-armstrong commented:
@dianakhuang Would you mind being a second pair of eyes on this? Thanks!

andy-armstrong commented:
jenkins run lettuce

dianakhuang commented:
How does this relate to `edx-lint`? I wonder if that's why the .pep8 file didn't exist already.

benpatterson commented:
oh noes @andy-armstrong I didn't notice that it was the .pep8 file. This kind of stuff can live in setup.cfg. Can you swing that kind of change?

andy-armstrong commented:
Thanks both of you. I didn't know about setup.cfg. Do I just make my changes there (after I remove the .pep8 file) and then it will just work? I guess I'll find out.

benpatterson commented:
@andy-armstrong yep. That should do it. In fact, I see .pycharm_helpers is already ignored. So hopefully when you are running pep8 locally you're already not-getting that directory (assuming it's at proj root).

andy-armstrong commented:
@benpatterson @dianakhuang Here's an updated version that uses setup.cfg instead of adding a new .pip8 file.

benpatterson commented:
:+1:

dianakhuang commented:
:+1: from me.
